lib/srv/forward/sshserver.go

```diff
@@ -603,15 +603,13 @@ func (s *Server) Serve() {
 		}
 
 		if s.targetServer.GetSubKind() == types.SubKindOpenSSHEICENode {
-			openTunnelResp, err := s.setupTunnelForOpenSSHEphemeralNode(ctx)
+			sshSigner, err := s.sendSSHPublicKeyToTarget(ctx)
 			if err != nil {
-				s.log.Warnf("Unable to set up EC2 Instance Connect Endpoint Tunnel for %q: %v", s.targetServer.GetName(), err)
+				s.log.Warnf("Unable to upload SSH Public Key to EC2 Instance  %q: %v", s.targetServer.GetName(), err)
```

Comment:
Does any error get returned to the client in this case? Should we reject the channel request here with an error message?

Reply:
We get the following log
```
2023-08-31T09:14:57Z WARN [NODE:FORW] Unable to upload SSH Public Key to EC2 Instance  "109ad5ab-7b9a-4dc6-9ae5-2a8802422490": send ssh public key failed for instance i-071fe80a271e646ab: operation error EC2 Instance Connect: SendSSHPublicKey, https response error StatusCode: 400, RequestID: 9cc598d9-1fa6-4c5c-a538-f9c5879b060d, api error AccessDeniedException: User: arn:aws:sts::278576220453:assumed-role/MarcoTestRoleOIDCProvider/1693473297415330025 is not authorized to perform: ec2-instance-connect:SendSSHPublicKey on resource: arn:aws:ec2:eu-west-2:278576220453:instance/i-071fe80a271e646ab because no identity-based policy allows the ec2-instance-connect:SendSSHPublicKey action trace.fields:map[dst-addr:172.31.32.234:22 src-addr:92.250.90.215:42576] forward/sshserver.go:608
```
And the client gets the following
![image](https://github.com/gravitational/teleport/assets/689271/8d0c9b28-d6b6-4b94-8930-1e646877f9a0)

Can we change this, tho?

I tried to do `s.rejectChannel(chans, "<error message>")` but it does not send anything back :thinking: 
For example, if I enable moderated sessions for the role I get the following error instead of the expected message
![image](https://github.com/gravitational/teleport/assets/689271/b620cdca-bbdd-46d8-96bd-bc2f7b3306eb)


On a side note, I improved the error message when creating a tunnel returns an error and that will look like this (truncated)
Previously it would only show `websocket bad handshake`
![image](https://github.com/gravitational/teleport/assets/689271/d0023809-8814-441b-99e3-ebf4c9f6c3f0)